README
This is the Perl distribution Google-Chart.

INSTALLATION

Google-Chart installation is straightforward. If your CPAN shell is
set up, you should just be able to do

    % cpan Google::Chart

Download it, unpack it, then build it as per the usual:

    % perl Makefile.PL
    % make && make test

Then install it:

    % make install

DOCUMENTATION

Google-Chart documentation is available as in POD. So you can do:

    % perldoc Google::Chart

to read the documentation online with your favorite pager.

Marcel Gruenauer


Changes
Changes
======

0.09000_04 Nov 09 2009
  * rt #51225 t/108_env_proxy.t was failing when http_proxy was being 
    explicitly set.

0.09000_03 Nov 04 2009
  * rt #51094: Andreas Koenig informs be that some dependencies were missing.
    added namespace::clean, Encode, Scalar::Util, and Sub::Exporter.
    (The latter few should actually be there already because of other deps)

  * remove t/105_legend.t, as Google::Chart::Legend no longer exists

0.09000_02 Oct 30 2009
  * Apply some changes to the Pie chart (nudge and patch by Jason Benguerel).
    the chp parameter and cht=pc doesn't exist in the Japanese docs.
    Sigh...

  * Above change also means name of some attributes changed
    (pie_labels -> labels, etc)

0.09000_01 Oct 29 2009
  * So I'm being a bad boy and doing it again. Time for another take.
    This time I'd like everybody's input.

    RATIONALE: I wrote the code for Google::Chart (after having received
    maintainership from Marcel) when I was going crazy about Roles,
    but I can't say I had the appropriate knowledge to deal with them.
   
    In this release, I've fixed what I can, organized things a bit better,
    and conjured up what I believe to be a saner API.

  * Significant <BACKWARD IMCOMPATIBLE> Changes
    - New syntax

        my $g = Google::Chart->create(
          $type => %args
        );
        $g->add_dataset(...);

    - Most operations are now done against the main chart object via
      add_* calls:

        $g->add_axis( ... );
        $g->add_dataset( ... );
        $g->add_range_fill( ... );

    - Related to the above, since we now cover most of the gory details in
      method calls, lots of object deprecated, or made invisible to 
      regular users.
      Things like markers, fill objects, etc, are now hidden.

0.05014 - 21 Aug 2009
  * Having completely forgotten about releasing a prod release for
    4 months, I now version++ at the prodding of Adam Kennedy ;) 

0.05014_01 - 30 Apr 2009
  * Apply changes from Mike Schilli

  (ms) Added support for XY type graphs
  (ms) Added Google::Chart::Title support in Google::Chart
  (ms) Added Google::Chart::Margin support
  (ms) Google::Chart::Fill::LinearGradient now supports transitions from
       color1/offset1 to color2/offset2 
  (ms) Added 'D' option in Google::Chart::Marker for line and bar chart 
       line styles

  * TODO: Now that we (well, /I/) know more about meta programming, perhaps
    there was an easier way to add the various functionalities using Roles?

0.05013 - 30 Jan 2009
  * Explicitly require Test::More 0.82 (rt #42907)

0.05012 - 10 Jan 2009
  * Google::Chart::Data::Simple was missing a 'with()' call. 

0.05011 - 08 Dec 2008
  * Make sure to set binmode() for render_to_file(), which caused Win32
    people to not be able to read the rendered files. Reported by Ron Pero.

0.05010 - 26 Oct 2008
  * Fix tests to accomodate for stricter LWP::UserAgent
  * Don't fail tests when you don't have network connectivity
  * Make _build_ua() eplicitly private

0.05009 - 25 Oct 2008
  * Apply patch from Richard Faasen to read set env_proxy by default 

0.05008 - 09 Oct 2008
  * Fix doc/implementation mismatch rt #39889

0.05007 - 23 Sep 2008
  * No code change
  * Fix doc problem
    http://www.nntp.perl.org/group/perl.cpan.testers/2008/09/msg2267672.html

0.05006 - 16 Sep 2008
  * Fix rt #39317 by allowing parameter coercion to accept a plain
    hashref (as the synopsis says it does).

0.05005 - 10 Sep 2008
  * Switch to using auto_include_deps, which seems to fix rt #39103
  * Attempt to fix garbled UTF-8 encoding when UTF-8 encoded strings are
    passed to QRcode.pm (reported by gugod)

0.05004 - 08 Sep 2008
  * Add Grid chart (Nobuo Danjou)
  * Make it possible to work with negative values (Nobuo Danjou)
  * Support multiple markers (Nobuo Danjou)
  * Recent versions of Moose (properly) unimport blessed() and confess(), 
    which breaks my code depending on it.

0.05003 - 13 Aug 2008
  * Fix semantics for Bar chart (Nobuo Danjou)

0.05002 - 06 Aug 2008
  * Fix stupid mistake in Makefile.PL to allow a simple "y" for
    prompted question

0.05001 - 06 Aug 2008
  * Attempt to fix tests by using Test::UseAllModules (rt #38233)

0.05000 - 29 Jul 2008
  * Release new version. Please be careful as there are MANY backwards
    incompatible changes
  * Tweak Makefile.PL so it warns if you already have an old version
    installed.

0.05000_03 - 29 Jul 2008
  * Make Data::Extended capable of handling multiple sets of data (danjou)
  * Add proper color support and legend support (danjou)

0.05000_02 - 25 Jul 2008
  * Fix Fill::LinearStripes and Fill::LinearGradient
  * Slight modification Google::Chart accessor generation

0.05000_01 - 24 Jul 2008
  * Mooooooooooooooose
  * Backwards incompatible! Beware!

Previous versions
  * Much work done by Marcel Grünauer. Changes are not listed, because
    I was too lazy to convert YAML changes to plain text.



Makefile.PL
use strict;
use inc::Module::Install;

my $have_previous = eval {
    require Google::Chart;
    $Google::Chart::VERSION < 0.05000;
};

if ($have_previous) {
    my $y_n = prompt(<<EOM, "n");

You seem to have an old version of Google::Chart installed.
As this version of Google::Chart breaks a lot of backwards compatibility,
installing this *WILL* break your code if you have something that depends on it.

Are you sure you want to continue? [y/N]
EOM
    if ($y_n !~ /^y(?:es)?$/i) {
        exit 0;
    }
}

name 'Google-Chart';

all_from('lib/Google/Chart.pm');

requires 'namespace::clean';
requires 'Encode';
requires 'LWP::UserAgent';
requires 'Moose', '0.92';
requires 'Sub::Exporter';
requires 'Scalar::Util';
requires 'URI';
requires 'URI::Escape';

test_requires 'Test::MockObject';
test_requires 'Test::More' => '0.82';
test_requires 'Test::UseAllModules';
no_index directory => 'docs';
no_index directory => 'tool';

# auto_include_deps;
WriteAll;
